### DISCUSSION TITLE
Hi I am just getting started with NocoDB [Feb 08]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5113] [Discord Threads]
> 🙋‍♂️ **Joecode**

**Comment by o1lab** [comment]
> 🙋‍♂️ **Joecode**  
#### Here are the commands I'm running:   
#### 1. cd packages/nocodb-sdk   
#### 2. npm install   
#### 3. npm run build   
#### 4. cd ../nc-gui   
#### 5. npm install   
#### 6. cd ../nocodb   
#### 7. EE=true npx webpack --config webpack.config.js   
#### 8. npx tsc -p tsconfig.json   
#### 9. npx tsc -p tsconfig.module.json   
#### 10. EE=\"true-xs-test\" npx webpack --config docker/wepack.config.js   
#### 11. docker build -t my-nocodb .   
####    
#### The image is built, but when I try and run it I get an error that module 'nocodb-sdk' cannot be found.  
#  
> 🙋‍♂️ **wkw**  
#### after step 5, u need to run `npm run generate`.    
#### after webpack, u need to unstall nocodb-sdk as it is packed with the build  
#  
> 🙋‍♂️ **Joecode**  
#### Thank you <@495466997304262658> . I tried running npm run build:copy after step 5, but I am still getting the 'cannot find module nocodb-sdk' message. Is there any other step I'm missing?  
#  
> 🙋‍♂️ **Joecode**  
#### <@495466997304262658> can you clarify how to install the nocodb-sdk into the build with docker?  
#  
> 🙋‍♂️ **wkw**  
#### sorry. missed your message. it should be included in nc-gui. after `npm run generate`, u need to run `rsync -rvzh ./dist/ ../nocodb/docker/nc-gui/`  
#  
> 🙋‍♂️ **Joecode**  
#### Thank you for your patients and explanation <@495466997304262658> . I am still having some trouble with this. Since by default nc-lib-gui comes from npm registry, why do we need to run rsync into `../nocodb/docker/nc-gui`?  Does the Dockerfile need to be modified to copy the module into the docker container? How does the sdk module get copied into the docker container? Thx  
#  
> 🙋‍♂️ **wkw**  
#### bc i assume you are using develop branch and build from local  
#  
> 🙋‍♂️ **wkw**  
#### since in develop branch, nocodb-sdk is pointing locally instead of npm package  
#  
> 🙋‍♂️ **wkw**  
#### npm run generate would package frontend to nc-gui-lib, here it should include the local nocodb-sdk  
#  
> 🙋‍♂️ **Joecode**  
#### yes correct I'm on develop branch, and the end goal is to have my local nc-gui and nocodb-sdk code bundled into my docker image. Is it webpack that is packaging up the modules or should they be copied into the docker image before running `npm ci`? With the existing Dockerfile it only copies package.json, package-lock.json and docker/main.js  
#  
> 🙋‍♂️ **wkw**  
#### oh in your dockerfile, u need to add    
####    
#### ```   
#### COPY ./docker/nc-gui/ ./docker/nc-gui/   
#### ```  
#  
> 🙋‍♂️ **Joecode**  
#### Thank you <@495466997304262658> !! I owe you a beer 🍺  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1060608046860095588)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
